tests/systems/renderer.test.js

```diff
@@ -41,11 +41,40 @@ suite('renderer', function () {
     document.body.appendChild(sceneEl);
   });
 
-  test('change renderer sortObjects', function (done) {
+  test('change renderer sortTransparentObjects', function (done) {
     var sceneEl = createScene();
-    sceneEl.setAttribute('renderer', 'sortObjects: true;');
+
```

Comment:
The fact that Unit Tests mock out the `setupRenderer` function means that I was limited in how much of the code changes I could Unit Test in a straightforward way.

Should we be looking for a way to improve UT coverage of the `setupRenderer` function?

Reply:
It seems that the configuration of the renderer is a bit split between both `setupRenderer` and the `init` function of the renderer system. It might be easier to move the setting of `sortObjects` and sorting functions into the `init` function as well. That way the tests here can:

- Verify that the default values are set properly (currently the test `'default initialization'` asserts that sortObjects defaults to false and doesn't fail) 
- Verify that changing the `sortTransparentObjects` result in the correct sorting functions being set

This unit test now also tests that the sorting _behaviour_ is correct, but I would split that off into a new `sortFunctions.test.js` that specifically tests them and let this unit test here cover only the responsibility of the renderer system of setting the right sort functions.

Finding a way to improve test coverage of `setupRenderer` would still be nice, but I think with the above the need for it in the scope of this change is mostly alleviated

Reply:
Pushed some changes to address these comments...

> It might be easier to move the setting of sortObjects and sorting functions into the init function as well.

The renderer `init` function will only be called if `renderer` system is initialized.  It does appear that's the case today, but I'm not sure it's good to bake that assumption into core code.  So I'd argue these properties should be set by the `AScene` class, and `setupRenderer()` seems like the right place to do it.

With the updated code, `renderer.sortObjects` is always `true`, and the `renderer` component has no bearing on it at all, so I'd argue it is cleaner to not have the `renderer` code touch `renderer.sortObjects`.

Sorting functions are set in the renderer `update` function, as well as in `setupRenderer()`.   No reason to move them to `init`, which just restricts them from being changed later.

> currently the test 'default initialization' asserts that sortObjects defaults to false and doesn't fail

Removed that check from the test.  `renderer.sortObjects` no longer has anything to do with the `renderer` component, so that check doesn't belong here any more.

> This unit test now also tests that the sorting behaviour is correct, but I would split that off into a new sortFunctions.test.js that specifically tests them and let this unit test here cover only the responsibility of the renderer system of setting the right sort functions.

That's fair.  Since the since the sort functions are internal to A-frame, I hadn't twigged that I can just import them, and then compare the functions directly.  Have now done that and it works fine.

A full UTs of the sort functions is also a sensible idea - I have added.

Reply:
> The renderer init function will only be called if renderer system is initialized. It does appear that's the case today, but I'm not sure it's good to bake that assumption into core code. So I'd argue these properties should be set by the AScene class, and setupRenderer() seems like the right place to do it.

I kind of see it the other way around. It's the AScene class that is effectively overstepping its boundaries by depending on both the `camera` and `renderer` systems. Conceptually in an ECS if the renderer system would be left out, you'd be left with a headless instance. From a pragmatic and legacy perspective I can see how this situation came to be and it's now unfeasible to every change (lots of code depends on the scene holding the renderer and camera).

That does leave us in the situation where renderer initialization is effectively split in two parts `AScene.setupRenderer` and `renderer.init`. The logical split I see at the moment is that `setupRenderer` only sets the properties needed at `WebGLRenderer` construction time and `renderer.init` sets anything that can change 'on the fly'. The object sorting is the only odd one out (even before this PR).

That being said, the actual `sortObjects` property can be left default (= true) so neither needs to assign anything. Unless we explicitly want to assert this default (and potentially shield us against three.js ever changing that default).

Reply:
Not sure I completely agree, as most of the `renderer` settings can't be changed on the fly (according to the docs, at least).

https://aframe.io/docs/1.4.0/components/renderer.html

But happy this is the right direction of travel, so I made the changes you suggested.

Reply:
Ah, you're right, seems I overlooked the few that are set by renderer but not changeable on the fly. The docs and the schema do paint a clear picture that all these settings do belong to the renderer system, yet a good chunk of them are handled (parsed even) by the `AScene`, which in an ideal setup wouldn't have been the case, but alas.

Anyway, the PR looks good. Just two more ES6 destructuring assignments in [renderer.test.js](https://github.com/aframevr/aframe/pull/5341/files#diff-4f5dc6eb19ddb970dd8594098448beffadc70d68317a24d00051a0d7682946b8R2-R4) and [sortFunctions.test.js](https://github.com/aframevr/aframe/pull/5341/files#diff-3029e71ddd0f460cba008f65476fe6b42a4ef168c7b1649b793b64f2e422d7edR3) and then it's all good to go IMO.

Reply:
My understanding was that the ES5 restriction only applies to /src.

Plenty of existing code under /test uses let, const, arrow functions etc. 

See e.g. [animation.test.js](https://github.com/aframevr/aframe/blob/master/tests/components/animation.test.js)

Reply:
Well, then everything is good to go IMO :-)